client/src/components/PublicPages/AuditPlanner/ElectionResultsCard.tsx

```diff
@@ -0,0 +1,368 @@
+                  <FormGroup
+                    helperText={
+                      <ErrorMessage
+                        errors={errors}
+                        name={`candidates[${i}].name`}
+                        render={({ message }) => message}
+                      />
+                    }
+                    intent={errors.candidates?.[i]?.name && 'danger'}
+                  >
+                    <InputFullWidth
+                      aria-label={`Candidate ${i} Name`}
+                      className={classnames(
+                        Classes.INPUT,
+                        errors.candidates?.[i]?.name && Classes.INTENT_DANGER
+                      )}
+                      defaultValue={candidateField.name}
+                      name={`candidates[${i}].name`}
+                      placeholder="Candidate name"
+                      readOnly={!editable}
+                      ref={register({
+                        required: 'Required',
+                      })}
+                    />
+                  </FormGroup>
+                </td>
+                <td>
+                  <FormGroup
+                    helperText={
+                      <ErrorMessage
+                        errors={errors}
+                        name={`candidates[${i}].votes`}
+                        render={({ message }) => message}
+                      />
+                    }
+                    intent={errors.candidates?.[i]?.votes && 'danger'}
+                  >
+                    <input
+                      aria-label={`Candidate ${i} Votes`}
+                      className={classnames(
+                        Classes.INPUT,
+                        errors.candidates?.[i]?.votes && Classes.INTENT_DANGER
+                      )}
+                      defaultValue={`${candidateField.votes}`}
+                      name={`candidates[${i}].votes`}
+                      onChange={validateAllCandidateVotesFields}
+                      placeholder="0"
+                      readOnly={!editable}
+                      ref={register({
+                        min: {
+                          value: 0,
+                          message: 'Cannot be less than 0',
+                        },
+                        required: 'Required',
+                        validate: () => {
+                          if (
+                            // No need to display this message for all candidate votes inputs
+                            i === 0 &&
+                            getValues().candidates.every(
+                              candidate => candidate.votes <= 0
+                            )
+                          ) {
+                            return 'At least 1 candidate must have greater than 0 votes'
+                          }
+                          return true
+                        },
+                        valueAsNumber: true,
+                      })}
+                      type="number"
```

Comment:
Is it possible for these number inputs to start empty instead of starting with 0? 0 doesn't seem like a useful default value since users will almost always need to change it.

Reply:
Good call - taken care of in https://github.com/votingworks/arlo/pull/1592/commits/1ec00130e75acae0001724b88433841fb2ea943e

Required tweaking types to allow `null` as a value for numeric inputs. While react-hook-form automatically allows all values to be `undefined`, I had to add and use `null` to get clearing of inputs via the "Clear" button to work